util/reggen/field.py

```diff
@@ -39,9 +39,31 @@
     'tags': [
         's',
         "tags for the field, followed by the format 'tag_name:item1:item2...'"
+    ],
+    'mubi': [
+        'b',
+        "boolean flag for whether the field is a multi-bit type"
     ]
 }
 
+# This information should be gathered directly from
+# the templating script later
+MUBI_TYPES = ['mubi4', 'mubi8', 'mubi12','mubi16']
+
+MUBI_VAL_TRUE = {
+    'mubi4': "0xa",
+    'mubi8': "0x5a",
+    'mubi12': "0xa5a",
+    'mubi16': "0x5a5a",
+}
+
+MUBI_VAL_FALSE = {
+    'mubi4': "0x5",
+    'mubi8': "0xa5",
+    'mubi12': "0x5a5",
+    'mubi16': "0xa5a5",
+}
```

Comment:
Can I suggest you zip these two? Also, no need for the "mubi" in the keys: this is just a width. And I'd probably not use a string for the values either. You'd end up with something like:
```python
# A dict mapping a MUBI width to a pair (m0, m1) where m0 is the value
# representing FALSE and m1 is the value representing TRUE.
MUBI_RAW_VALS = {
    4: (0x5, 0xa),
    8: (0xa5, 0x5a),
    12: (0x5a5, 0xa5a),
    16: (0x5a5a, 0xa5a5)
}
```
I've left notes below for how this could be used.

Reply:
Reading the last commit, I think it's probably better to get rid of these dicts entirely. Can't we just assert that the width in `get_mubi_val` is a multiple of 4 and then build up the value by hand. That also decouples this code from the prim, since we can support any old (nibble-aligned) width.